### PR TITLE
Users page fixes

### PR DESCRIPTION
**Pull request by marusak**

### CODE REVIEW

File: pkg/users/authorized-keys-panel.js [hunk 1]
```diff
@@ -91,73 +96,56 @@ export function AuthorizedKeys({ name, home, allow_mods }) {
...
-                    </div>
-                </li>);
-        }
-    } else
+    let error = "";
+
+    if (state == "access-denied")
+        error = _("You do not have permission to view the authorized public keys for this account.");
+    else if (state == "failed")
+        error = _("Failed to load authorized keys.");
```
Comment (cockpituous) [inline pkg/users/authorized-keys-panel.js:104]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18014-20221209-092338-d41e2802-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/authorized-keys-panel.js [hunk 2]
```diff
@@ -91,73 +96,56 @@
...
-                </Button>}
-            </div>
-            <div className="pf-c-card__body contains-list">
-                <ul className="pf-c-data-list pf-m-compact" id="account-authorized-keys-list">
-                    { key_items }
-                </ul>
-            </div>
-        </div>
+                <CardActions>
+                    <Button variant="secondary" id="authorized-key-add" onClick={() => add_authorized_key_dialog(manager)}>
```
Comment (cockpituous) [inline pkg/users/authorized-keys-panel.js:116]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18014-20221209-092338-d41e2802-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/authorized-keys-panel.js [hunk 3]
```diff
@@ -91,73 +96,56 @@
...
+                    columns={ [
+                        { title: _("Name"), header: true },
+                        { title: _("Fingerprint") },
+                        { title: _("Actions") },
+                    ] }
+                    rows={keys.map(k => {
+                        return {
+                            columns: [
+                                { title: k.comment || _("Unnamed"), props: { width: 20 } },
+                                { title: k.fp || _("Invalid key"), props: { width: 80 } },
```
Comment (cockpituous) [inline pkg/users/authorized-keys-panel.js:137]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18014-20221209-092338-d41e2802-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/authorized-keys-panel.js [hunk 4]
```diff
@@ -91,73 +96,56 @@
...
+                                {
+                                    title: <Button key={k.fp} variant="secondary" onClick={() => remove_key(k.raw)}>
```
Comment (cockpituous) [inline pkg/users/authorized-keys-panel.js:139]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18014-20221209-092338-d41e2802-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/authorized-keys-panel.js [hunk 5]
```diff
@@ -91,73 +96,54 @@ export function AuthorizedKeys({ name, home, allow_mods }) {
...
-            </div>
-        </div>
+                <CardActions>
+                    <Button variant="secondary" id="authorized-key-add" onClick={() => add_authorized_key_dialog(manager)}>
+                        {_("Add key")}
+                    </Button>
+                </CardActions>
+                }
+            </CardHeader>
+            <ListingTable
```
Comment (KKoukiou) [inline pkg/users/authorized-keys-panel.js:122]: This breaks mobile. I can help you if you want to make it work with this code.

Comment (marusak) [reply pkg/users/authorized-keys-panel.js:122]: Do we need some breakpoints there?

Comment (marusak) [reply pkg/users/authorized-keys-panel.js:122]: ah I see, I am taking a look

Comment (tomasmatus) [reply pkg/users/authorized-keys-panel.js:122]: I found two similar PF examples.  Basically they convert the button to kebab menu on mobile view. Maybe you could try it

previews:
https://www.patternfly.org/v4/components/table/react-composable/composable-actions-overflow/
https://www.patternfly.org/v4/components/table/react-composable/composable-actions/

code here: https://www.patternfly.org/v4/components/table/#composable-actions-overflow

Comment (KKoukiou) [reply pkg/users/authorized-keys-panel.js:122]: What @tomasmatus suggests or use DataList instead of ListingTable.

**Comment by marusak** [comment]
@tomasmatus and @KKoukiou I have changed it, now looks like this on iphone5s

Screenshot from 2022-12-13 11-28-19

File: pkg/users/authorized-keys-panel.js [hunk 6]
```diff
@@ -85,79 +92,94 @@ export function AuthorizedKeys({ name, home, allow_mods }) {
...
+                                                    {_("Remove")}
+                                                </Button>
+                                            </OverflowMenuItem>
+                                        </OverflowMenuGroup>
+                                    </OverflowMenuContent>
+                                    <OverflowMenuControl>
+                                        <Dropdown position="right"
+                                                  onSelect={() => {
+                                                      if (openedMenu.indexOf(k.fp) >= 0)
+                                                          setOpenedMenu(openedMenu.filter(m => m !== k.fp));
```
Comment (cockpituous) [inline pkg/users/authorized-keys-panel.js:155]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18014-20221213-102921-2058c207-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/authorized-keys-panel.js [hunk 7]
```diff
@@ -85,79 +92,94 @@
...
+                                                      else
+                                                          setOpenedMenu([...openedMenu, k.fp]);
+                                                  }}
```
Comment (cockpituous) [inline pkg/users/authorized-keys-panel.js:158]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18014-20221213-102921-2058c207-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/authorized-keys-panel.js [hunk 8]
```diff
@@ -85,79 +92,94 @@
...
+                                                  toggle={
+                                                      <KebabToggle
+                                                      onToggle={open => {
+                                                          if (open)
+                                                              setOpenedMenu([...openedMenu, k.fp]);
```
Comment (cockpituous) [inline pkg/users/authorized-keys-panel.js:163]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18014-20221213-102921-2058c207-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/authorized-keys-panel.js [hunk 9]
```diff
@@ -85,79 +92,94 @@
...
+                                                          else
+                                                              setOpenedMenu(openedMenu.filter(m => m !== k.fp));
+                                                      }}
```
Comment (cockpituous) [inline pkg/users/authorized-keys-panel.js:166]: These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18014-20221213-102921-2058c207-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/users/authorized-keys-panel.js [hunk 10]
```diff
@@ -85,79 +92,94 @@
...
+                                                      />
+                                                  }
+                                                  isOpen={openedMenu.indexOf(k.fp) >= 0}
+                                                  isPlain
+                                                  dropdownItems={[<OverflowMenuDropdownItem key="delete" isShared onClick={() => remove_key(k.raw)}>
```
Comment (cockpituous) [inline pkg/users/authorized-keys-panel.js:171]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18014-20221213-102921-2058c207-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by KKoukiou — APPROVED** [review]

**Pull request merged**